frontend/public/components/operator-hub/operator-hub-subscribe.tsx

```diff
@@ -97,18 +124,69 @@ export const OperatorHubSubscribeForm = withFormState((props: OperatorHubSubscri
     };
 
     return (!_.isEmpty(catalogSourceConfig) || hasBeenEnabled
-      ? k8sUpdate(CatalogSourceConfigModel, {...catalogSourceConfig, spec: {targetNamespace: operatorGroupNamespace, packages}}, 'openshift-marketplace', OPERATOR_HUB_CSC_NAME)
+      ? k8sUpdate(CatalogSourceConfigModel, {...catalogSourceConfig, spec: {...catalogSourceConfig.spec, packages}}, 'openshift-marketplace', OPERATOR_HUB_CSC_NAME)
       : k8sCreate(CatalogSourceConfigModel, newCatalogSourceConfig)
-    ).then(() => k8sCreate(SubscriptionModel, subscription))
+    ).then(() => props.operatorGroup.data.some(group => group.metadata.namespace === props.formState().targetNamespace)
```

Comment:
Create `OperatorGroup` (if it doesn't already exist) which enables the target namespace for OLM.